src/jit/codegenxarch.cpp

```diff
@@ -7890,7 +7890,90 @@ CodeGen::getBaseVarForPutArgStk(GenTreePtr treeNode)
     {
         // The first baseVarNum is guaranteed to be the first incoming arg of the method being compiled.
         // See lvaInitTypeRef() for the order in which lvaTable entries are initialized.
+        // For Windows the value is always 0 (there are always slots in the caller's frame for the first 4 args to a callee).
+        // For System V the first stack arg should be calculated.
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        baseVarNum = 0;
+        CORINFO_ARG_LIST_HANDLE argLst = compiler->info.compMethodInfo->args.args;
+        const unsigned argSigLen = compiler->info.compMethodInfo->args.numArgs;
+        
+        unsigned usedIntRegs = 0;
+        unsigned usedFloatRegs = 0;
+        
+        for (unsigned i = 0;
+        i < argSigLen;
+            i++, argLst = compiler->info.compCompHnd->getArgNext(argLst))
+        {
+            LclVarDsc* varDsc = &(compiler->lvaTable[i]);
```

Comment:
Why is it not possible to simply check lvArgReg? Shouldn't it be the case that the first variable that has lvArgReg == REG_STK would be the one you are interested in?